pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -371,18 +373,24 @@ def on_validation_end(self, trainer, pl_module):
             elif self.check_monitor_top_k(current):
                 self._do_check_save(filepath, current, epoch, trainer, pl_module)
             elif self.verbose > 0:
-                log.info(f'\nEpoch {epoch:05d}: {self.monitor}  was not in top {self.save_top_k}')
+                log.info(f'Epoch {epoch:d}: {self.monitor} was not in top {self.save_top_k}')
 
         else:
             if self.verbose > 0:
-                log.info(f'\nEpoch {epoch:05d}: saving model to {filepath}')
+                log.info(f'Epoch {epoch:d}: saving model to {filepath}')
 
             assert trainer.global_rank == 0, 'tried to make a checkpoint from non global_rank=0'
             self._save_model(filepath, trainer, pl_module)
 
         if self.save_last:
-            filepath = os.path.join(self.dirpath, self.prefix + ModelCheckpoint.CHECKPOINT_NAME_LAST)
+            filename = self._format_checkpoint_name(
+                self.CHECKPOINT_NAME_LAST, epoch, ckpt_name_metrics, prefix=self.prefix
+            )
```

Comment:
what will the last be formated?
The good thing about having it just named "last.ckpt" was that you could easily load programmatically the last checkpoint without knowing the epoch. It would be great if we could keep that.

Reply:
To allow having the epoch (and any other metrics) in the last checkpoint name is one of the main features of this PR. If the user has not modified `CHECKPOINT_NAME_LAST` the last will just be named `last.ckp`. Just as before.

Reply:
very nice!